messaging/publisher/persistence.go

```diff
@@ -5,8 +5,10 @@ import (
 )
 
 type Persistence interface {
-	Get() (int64, error)
-	Set(int64) error
+	GetLastPublished() (int64, error)
+	SetLastPublished(int64) error
+	GetLastAcked(identity []byte) (int64, error)
```

Comment:
Do we ever delete acks? If after `ErrDeviceNotFound`, we receive a message from the same identity including our device, I think we shouldn't send them an empty message periodically anymore. Also, does not it seem like a duplication of some logic in contact code where we publish information about our bundles/devices?

Reply:
We never delete acks, but we only reply with an Ack when we have `ErrDeviceNotFound`, it's not sent periodically.

It's similar in purpose to SendContactCode, but it's only triggered when we know that the other user does not know about our device (or has a stale list), it offers some redundancy as we can't make much guarantees on whisper/mailservers.

Basically it works this way:

1) I send you a message not including your device -> You reply with an Ack letting me know
2) I send you another message not including your device -> If the cooldown period has passed (2 hours), I send back an ack, otherwise nothing
3) I send you a message including your device -> Nothing is sent back
4) I send you a message not including your device -> same as point 2

Reply:
> We never delete acks, but we only reply with an Ack when we have ErrDeviceNotFound, it's not sent periodically.

Ah right. Not sure it stuck in my head it's periodical. 

Yeah, the flow is clear. Question is whether we need a table to stor this information or it could be in memory 🤔 Possibly someone could create new identities and keep forcing us to send acks over and over and filling up the database as it's never cleared up.

Reply:
That's a good idea, although having it in memory might trigger a lot of sending messages on startup, when we retrieve data from the mailserver, but it is an option ( I would use a table for now just to be on the safe side).

The attack over acks is a concern, but in that case is much simpler for an attacker to send a legit message each time with a new identity, and fill that database up, as we would save the new messages. 

I would keep it as it for now not to introduce too many changes (status-react currently uses the same method), but I will add a comment, I think it's a good idea to store in memory (and possibly just rate-limit the sending to avoid the scenario when we start the app), and I will integrate this with status-react and next week I'll add a inmemory persistence service and we can test things out, what do you reckon?

Reply:
actually the more I think of it , the more I think you are right and should be in memory, so I will change it next week, contact-code as well, no reason not to be as well, thanks for the suggestion!

Reply:
That sounds good! 👍

Reply:
Updated, I have only left `lastAcked` in memory, so we can try it out for a while, and if it works fine, I'll remove as well `lastPublished`. thanks for the suggestion